### ISSUE TITLE
Remove expression UUIDs from metadata section of a source file

### ISSUE THREAD

**Issue by 4e6** [enso-org/enso#10182]
### Goal

- #9257

This ticket describes the second part of  #9257 tasks. The metadata section currently contains two types of information: 
- node metadata like position, color,  and/or widget metadata
- identification of expressions (and nodes) for communication with the engine via the LS protocol

The goal of this issue is to deal **only with UUIDs of expressions** where _persistence isn't needed_. The goal isn't to remove the whole `META-DATA` concept, but make the section **significantly smaller**. 

### Current Situation

The rumor has it that for a less than hundred lines `.enso` file the size of `META-DATA` section can grow significantly over 100KB - mostly because of storing UUIDs for **all the expressions** in the program, not just UUIDs for `operatorXyz` assignments.

### The Plan

To [quote 9257 comment](https://github.com/enso-org/enso/issues/9257#issuecomment-1978474823):

> Once #7954 is implemented ... then the _Y.js based  AST_ is hosted in the same process as the engine ... when there is an AST change that _modifies the source code_ we **traverse the AST and recompute ranges** (offset, line, column) of **expressions and nodes** that _have an UUIDs attribute in the AST. A change in the source code always comes with new ranges. All of that is done **synchronously**, keeping _UUIDs and their spans_ in sync.

```[tasklist]
### Tasks
- [ ] traverse the AST and recompute ranges for elements with UUID attribute - let's call it **dynamic UUID range information**
- [ ] provide the _dynamic UUID range information_ to engine together with the source code change (probably by extending the LS TextEdit message)
- [ ] use the _dynamic UUID range information_ when parsing or while performing `TreeToIr` together with  `META-DATA` section in the source file - e.g. merge those two
- [ ] modify IDE to _avoid storing UUIDs_ of **expressions** in the `META-DATA` section and just provide them via _dynamic UUID range information_
- [ ] identify expressions in messages between GUI and Engine by both sources of _UUIDs_ - e.g. by `META-DATA` section and _dynamic UUID range information_
```

### TL;DR

- use _dynamic UUID range information_ where persistence is not needed
- keep persistent information (where visualization is attached, what is the type of expression, etc.) in the `META-DATA` section as so far
